eth/chains/base.py

```diff
@@ -314,7 +314,17 @@ def get_ancestors(self, limit: int, header: BlockHeaderAPI) -> Tuple[BlockAPI, .
         # We construct a temporary block object
         vm_class = self.get_vm_class_for_block_number(header.block_number)
         block_class = vm_class.get_block_class()
-        block = block_class(header=header, uncles=[], transactions=[])
+
+        if hasattr(block_class, "withdrawals"):
```

Comment:
**Note for refactoring:** This was made to work, re-think this logic a bit. Not just here, but this sort of logic appears in a few other places.